### PR TITLE
fix: infinitely get pieces when piece num is invalid

### PR DESCRIPTION
**Pull request by jim3ma**
Signed-off-by: Jim Ma <majinjing3@gmail.com>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [ ] Code compiles correctly.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by gaius-qi — DISMISSED** [review]
LGTM

**Review by 244372610 — DISMISSED** [review]

**Review by gaius-qi — DISMISSED** [review]
LGTM

File: client/daemon/peer/peertask_base.go [hunk 1]
```diff
@@ -816,24 +829,42 @@ func (pt *peerTask) getPieceTasks(span trace.Span, curPeerPacket *scheduler.Peer
...
+				pt.Warnf("get piece tasks with error: %s, but peer packet changed, switch to new peer packet, current destPeer %s, new destPeer %s", getError,
 					curPeerPacket.MainPeer.PeerId, lastPeerPacket.MainPeer.PeerId)
 				peerPacketChanged = true
 				return nil, true, nil
 			}
-			return nil, true, getErr
+			return nil, true, getError
 		}
+		// need update metadata
+		if piecePacket.ContentLength > pt.contentLength.Load() || piecePacket.TotalPiece > pt.totalPiece || uint32(piecePacket.TotalPiece) <= request.StartNum {
```
Comment (244372610) [inline client/daemon/peer/peertask_base.go:860]: 是否包含等于的情况 >=

**Review by 244372610 — APPROVED** [review]

File: client/daemon/peer/peertask_base.go (hunk 1 above)
Comment (jim3ma) [reply client/daemon/peer/peertask_base.go:860]: 不需要

**Review by 244372610 — APPROVED** [review]

**Review by gaius-qi — APPROVED** [review]
LGTM

**Pull request merged**